gulpfile.js

```diff
@@ -34,6 +34,22 @@ gulp.task('build_server_with_binary', function(done) {
 	done();
 });
 
+gulp.task('prepare_pre_release', function(done) {
+    const json = JSON.parse(fse.readFileSync("./package.json").toString());
+    const stableVersion = json.version.match(/(\d+)\.(\d+)\.(\d+)/);
+    const major = stableVersion[1];
+    const minor = stableVersion[2];
+    const date = new Date();
+    const month = date.getMonth() + 1;
+    const day = date.getDate();
+    const hours = date.getHours();
+    const patch = `${date.getFullYear()}${prependZero(month)}${prependZero(day)}${prependZero(hours)}`;
+    const insiderPackageJson = Object.assign(json, {
+        version: `${major}.${minor}.${patch}`,
```

Comment:
I think it might be a good idea to preserve the patch number from the `package.json`, and attach the build time info afterwards with a `+`. eg. `0.21.1+2022070614`  See https://semver.org/#spec-item-10

Reply:
Roland informed me this isn't supported https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions

Reply:
I think we should follow https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions since for now it seems they support strictly `major.minor.patch`